server/embed/config.go

```diff
@@ -907,6 +957,7 @@ func (cfg *Config) Validate() error {
 	if err := cfg.setupLogging(); err != nil {
 		return err
 	}
+	cfg.ServerFeatureGate.(featuregate.MutableFeatureGate).SetLogger(cfg.logger)
```

Comment:
Why this cannot be done in constructor?

Reply:
because the server logger is not set up before this point. And the ServerFeatureGate could be needed before that, for example there could be a feature gate to change the logger.